### DISCUSSION TITLE
Who.Type Task Where Clause not Working in mdt rollup

### DISCUSSION THREAD

**Discussion by Dustin-Northcutt** [jamessimone/apex-rollup#633] [Q&A] [answered]
I'm attempting to rollup Task data to Contacts and want to add `Who.Type = 'Contact'` to the Child Object Where Clause. When I attempt this I get the following error:
SObject row was retrieved via SOQL without querying the requested field: Task.Who

Here's my mdt record
![Screenshot 2024-10-03 at 1 55 01 PM](https://github.com/user-attachments/assets/f0f7e3fb-9646-469a-929f-56240480cf8e)

Guessing I'm doing something wrong?

**Comment by jamessimone** [comment]
Have you tried omitting that (e.g. using `ActivityDate <= TODAY AND ActivityDate >= LAST_N_DAYS:45 AND IsClosed = true AND TaskSubtype = 'Call'` ? Since you're not starting this rollup from the parent, I don't think you actually need to filter on the type since only matching Contacts will end up getting rolled up values assigned

**Reply by Dustin-Northcutt** [reply]
@jamessimone I have tried that and it is working as expected! Apologies I should have clarified that. I was thinking that perhaps the Who.Type filter would improve the query's performance but it may not matter.

**Reply by jamessimone** [reply, marked as answer]
No worries - I will have a look at this, though, to see what the underlying issue is. I have a few tests for validating `Who.Type` where clauses work but this must be some other part of the codebase or a gap I hadn't anticipated

**Reply by Dustin-Northcutt** [reply]
Thanks so much @jamessimone!
